### PR TITLE
Add `display-name` option to customize/disambiguate output

### PR DESCRIPTION
**Pull request by awf**
This PR introduces an optional `display-name` field in the repo configuration, which allows the user to override the default `parent/repo` format.  This is useful, for example, when indexing the same repo with different config options (e.g. main branch and feature branch).

E.g. this config file:
```
{
  "dbpath" : "db",
  "vcs-config" : {
    "git": {
      "ref" : "main"
    }
  },
  "repos" : {
    "Hound" : {
      "url" : "https://github.com/awf/hound.git"
    },
    "Hound/alpine" : {
      "url" : "https://github.com/hound-search/hound.git",
      "display-name" : "Hound (alpine)",
      "vcs-config" : {
        "git": {
          "ref" : "alpine"
        }
      }
    }
  }
}
```
produces this output (the red text) from the CLI client
image

and this on HTML
image

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**The PR fulfills these requirements:**
- [x] All tests are passing?
- [x] New/updated tests are included?
- [x] If any static assets have been updated, has ui/bindata.go been regenerated?
- [x] Are there doc blocks for functions that I updated/created?

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (salemhilal) [inline CONTRIBUTING.md:15]: 😅 Thank you!

**Review by salemhilal — APPROVED** [review]
This seems easy and reasonable! Let me get approval from some of the other maintainers before merging it. Thank you for contributing!

File: CONTRIBUTING.md
Comment (awf) [reply CONTRIBUTING.md:15]: np :)

File: config-example.json
Comment (awf) [reply config-example.json:19]: Just an example, because this view of this repo has exclude-dot-files=true.
But maybe it should just be something more obviously displayish, e.g. "RepoOne (dot files not indexed)"?

Comment (salemhilal) [reply config-example.json:19]: Ah, that makes sense. I agree with that rephrasing. If you don't mind updating it, I'll merge this PR.

Comment (awf) [reply config-example.json]: Done, thanks!

**Review by jvt — APPROVED** [review]
This looks great, awesome quality-of-life improvement! Thank you for contributing!

**Review by gmcaguilar — APPROVED** [review]

**Pull request merged**
